docs/user-guide/deployments.md

```diff
@@ -0,0 +1,284 @@
+        ports:
+        - containerPort: 80
+```
+
+[Download example](new-nginx-deployment.yaml?raw=true)
+<!-- END MUNGE: EXAMPLE new-nginx-deployment.yaml -->
+
+
+```console
+$ kubectl apply -f ./new-nginx-deployment.yaml
+deployment "nginx-deployment" configured
+```
+
+Running a get immediately will still give:
+
+```console
+$ kubectl get deployments
+NAME               UPDATEDREPLICAS   AGE
+nginx-deployment   3/3               8s
+```
+
+This indicates that deployment status has not been updated yet (it is still
+showing old status).
+Running a get again after a minute, will give:
+
+```console
+$ kubectl get deployments
+NAME               UPDATEDREPLICAS   AGE
+nginx-deployment   1/3               1m
+```
+
+This indicates that deployment has updated one of the three pods that it needs
+to update.
+Eventually, it will get around to updating all the pods.
+
+```console
+$ kubectl get deployments
+NAME               UPDATEDREPLICAS   AGE
+nginx-deployment   3/3               3m
+```
+
+We can run ```kubectl get rc``` to see that deployment updated the pods by creating a new RC
+which it scaled up to 3 and scaled down the old RC to 0.
+
+```console
+kubectl get rc
+CONTROLLER                CONTAINER(S)   IMAGE(S)      SELECTOR                                                         REPLICAS   AGE
+deploymentrc-1562004724   nginx          nginx:1.9.1   deployment.kubernetes.io/podTemplateHash=1562004724,app=nginx   3          5m
+deploymentrc-1975012602   nginx          nginx:1.7.9   deployment.kubernetes.io/podTemplateHash=1975012602,app=nginx   0          7m
+```
+
+Running get pods, will only show the new pods.
+
+```console
+kubectl get pods
+NAME                            READY     STATUS    RESTARTS   AGE
+deploymentrc-1562004724-0tgk5   1/1       Running   0          9m
+deploymentrc-1562004724-1rkfl   1/1       Running   0          8m
+deploymentrc-1562004724-6v702   1/1       Running   0          8m
+```
+
+Next time we want to update pods, we can just update the deployment again.
+
+Deployment ensures that not all pods are down while they are being updated. By
+default, it ensures that minimum of 1 less than the desired number of pods are
+up. For example, if you look at the above deployment closely, you will see that
+it first created a new pod, then deleted some old pods and created new ones. It
+does not kill old pods until a sufficient number of new pods have come up.
+
+```console
+$ kubectl describe deployments
+Name:                           nginx-deployment
+Namespace:                      default
+CreationTimestamp:              Thu, 22 Oct 2015 17:58:49 -0700
+Labels:                         app=nginx-deployment
+Selector:                       app=nginx
+Replicas:                       3 updated / 3 total
+StrategyType:                   RollingUpdate
+RollingUpdateStrategy:          1 max unavailable, 1 max surge, 0 min ready seconds
+OldReplicationControllers:      deploymentrc-1562004724 (3/3 replicas created)
```

Comment:
You should explicitly say that Deployment creates a new ReplicationController each time you update it, and then deletes them once those old versions are no longer needed, because they are replaced by a new version, or when they complete their rollout.  Also talk about there being multiple concurrently active rolling updates, and whether it always goes through all the updates it is given one by one, or whether it skips to the newest one.